src/main/java/io/jenkins/plugins/sprp/impl/AgentGenerator.java

```diff
@@ -0,0 +1,98 @@
+package io.jenkins.plugins.sprp.impl;
+
+import hudson.Extension;
+import io.jenkins.plugins.sprp.PipelineGenerator;
+import io.jenkins.plugins.sprp.models.Agent;
+import org.jenkinsci.Symbol;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Extension
+@Symbol("agent")
+public class AgentGenerator extends PipelineGenerator<Agent> {
+
+    @Override
+    public List<String> toPipeline(Agent agent) {
+        ArrayList<String> agentLines = new ArrayList<>();
+
+        if (agent == null) {
+            agentLines.add("agent any");
+        } else if (agent.getAnyOrNone() != null)
+            agentLines.add("agent " + agent.getAnyOrNone());
+        else if(agent.isNone()) {
```

Comment:
I'm still not sure why we have 2 checks here.  `getAnyOrNone()` literally returns `any` or `none`.  We should never reach this last check.